DOC: shared-libraries: fix and clarify install location example

AndrewAnnex opened the pull request:
Updates install_path to install_dir. 
Updates code block to be more explicit with shared_library usage.

Fixes #769

dnicolodi commented on docs/how-to-guides/shared-libraries.rst:
```diff
@@ -64,11 +64,16 @@ A shared library produced by ``library()`` or ``shared_library()`` built like th
 is installed to ``libdir`` by default. If the only reason the shared library exists
 is to be used inside the Python package being built, then it is best to modify
-the install location to be within the Python package itself:
+the install location, via the `install_dir` argument, to be within the Python package itself:
```
reST requires double backticks to format something as code. Can you also line wrap this at 78-80 characters?

dnicolodi commented on docs/how-to-guides/shared-libraries.rst:
```diff
@@ -64,11 +64,16 @@ A shared library produced by ``library()`` or ``shared_library()`` built like th
+        'example',
+        'examplelib.c',
+        install: true,
+        install_dir: py.get_install_dir() / 'mypkg/subdir',
```
I've just realized that this example and other in this file uses the Meson's `python` module imported as `py`. However, the example just below uses `py3`. While at this, can you please fix this inconsistency too (in a separate commit maybe)?

dnicolodi reviewed (changes requested):
Can you also please fix the commit message to comply with the format we use for this project? Namely, it should have a summary line of less than 78 characters, starting with an identifier in all caps, `DOC` in this case, followed by colons and the summary of the changes. The details go in the body. Something like
```
DOC: shared-libraries: fix and clarify install location example

Fixes #769
```

AndrewAnnex commented:
@dnicolodi happy to fix all the above, although for the commit message wouldn't that be easier to fix when squashing the commit on merge? I guess I can fix all the above, squash the commits, and force push to the branch too but that is a bit more involved

AndrewAnnex commented on docs/how-to-guides/shared-libraries.rst:
```diff
@@ -64,15 +64,16 @@ A shared library produced by ``library()`` or ``shared_library()`` built like th
         'examplelib.c',
         install: true,
-        install_dir: py.get_install_dir() / 'mypkg/subdir',
+        install_dir: py3.get_install_dir() / 'mypkg/subdir',
```
oh heh oops will do

dnicolodi commented:
> I guess I can fix all the above, squash the commits, and force push to the branch too but that is a bit more involved

You should do just that. The goal is to teach how to contribute to this project while freeing the maintainers from menial task.

dnicolodi reviewed (changes requested):
CI is still not happy.

dnicolodi commented:
I would use `text` for the code block language and leave a comment that points to the pygments issue.

AndrewAnnex commented:
okay cleaned up the commits, updated the code-block to text, added some comments about the pygments issue, and ran precommit locally to hopefully avoid further ci complaints

dnicolodi commented:
Thanks.
